### PR TITLE
`zk config` validation.

### LINKED ISSUE #421: Only generate keypair and write deploy alias to config if URL validates
### Describe the problem

If you provide an invalid GraphQL API URL during the creation of a deploy alias via `zk config`, a keypair is generated and written to the `keys` dir and the alias is added to `config.json`.

### Describe the proposed solution

Make keypair generation and writing the alias to `config.json` contingent on the validation of the URL

### Alternatives considered

_No response_

### Importance

would make my life easier

### Additional Information

_No response_

### LINKED ISSUE #428: Missing GraphQL URL and Fee amount validation for `zk config`
### Describe the bug

Missing GraphQL URL and Fee amount for `zk config`

### Steps to Reproduce

Steps:

- Issue `zk config` command;
- Provide invalid URL and/or whitespaces chars as the fee amount when asked.

As the result:
- Invalid URL validation message will be thrown in the end of the process (not at the time of entering the data). But new config will be saved anyway;
- There is no validation for Fee amount if whitespaces chars provided.

```shell
Enter values to create a deploy alias:
✔ Create a name (can be anything): · new local
✔ Set the Mina GraphQL API URL to deploy to: · q
✔ Set transaction fee to use when deploying (in MINA): ·
✔ Choose an account to pay transaction fees: · Use stored account q (public key: B62qpUqPcNSebuXuNhkm6Vf68fkKSAcwPPHkNN7GhhQV42jrapD5vsY)
prompt response {
  deployAliasName: 'new-local',
  url: 'q',
  fee: '   ',
  feepayer: 'defaultCache'
}
✔ Use stored fee payer q (public key: B62qpUqPcNSebuXuNhkm6Vf68fkKSAcwPPHkNN7GhhQV42jrapD5vsY)
✔ Create zkApp key pair at keys/new-local.json
✔ Add deploy alias to config.json
zk config

Add a new deploy alias

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]

TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:399:5)
    at new URL (node:internal/url:560:13)
    at getExplorerName (/Users/shymkiv/projects/o1/zkapp-cli/src/lib/config.js:332:10)
    at Object.config [as handler] (/Users/shymkiv/projects/o1/zkapp-cli/src/lib/config.js:232:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  input: 'q',
  code: 'ERR_INVALID_URL'
}

cat config.json
{
  "version": 1,
  "deployAliases": {
    "new-local": {
      "url": "q",
      "keyPath": "keys/new-local.json",
      "feepayerKeyPath": "/Users/shymkiv/.cache/zkapp-cli/keys/q.json",
      "feepayerAliasName": "q",
      "fee": "   "
    }
  }
}


--- 


Enter values to create a deploy alias:
✔ Create a name (can be anything): · local
✔ Set the Mina GraphQL API URL to deploy to: · http://p42vinga1:8080/graphql
✔ Set transaction fee to use when deploying (in MINA): ·
✔ Choose an account to pay transaction fees: · Create a new fee payer key pair
✔ Create an alias for this account · q
prompt response {
  deployAliasName: 'local',
  url: 'http://p42vinga1:8080/graphql',
  fee: '   ',
  feepayer: 'create',
  feepayerAliasName: 'q'
}
✔ Create fee payer key pair at /Users/shymkiv/.cache/zkapp-cli/keys/q.json
✔ Create zkApp key pair at keys/local.json
✔ Add deploy alias to config.json
feepayerKeyPair {
  privateKey: 'EKEcpFmgU9KZDFFC7jSYZwf1YSiK7Tfc88yPaCE8nd6Wp2bzHMRJ',
  publicKey: 'B62qpUqPcNSebuXuNhkm6Vf68fkKSAcwPPHkNN7GhhQV42jrapD5vsY'
}

Success!

Next steps:
  - If this is a testnet, request tMINA at:
    https://faucet.minaprotocol.com/?address=B62qpUqPcNSebuXuNhkm6Vf68fkKSAcwPPHkNN7GhhQV42jrapD5vsY&?explorer=undefined
  - To deploy, run: `zk deploy local`


cat config.json
{
  "version": 1,
  "deployAliases": {
    "local": {
      "url": "http://p42vinga1:8080/graphql",
      "keyPath": "keys/local.json",
      "feepayerKeyPath": "/Users/shymkiv/.cache/zkapp-cli/keys/q.json",
      "feepayerAliasName": "q",
      "fee": "   "
    }
  }
}
```

### System Info

```shell
Any
```

### Severity

annoyance

### Additional Information

_No response_

### LINKED ISSUE #462: No whitespaces-only validation for the Fee Payer alias
During the new Fee Payer creation or recovering one from private key there is no validation if alias name is represented with whitespaces only. 

```shell
cat config.json
{
  "version": 1,
  "deployAliases": {
    "1": {
      "url": "http://localhost:8282/graphql",
      "keyPath": "keys/1.json",
      "feepayerKeyPath": "/Users/shymkiv/.cache/zkapp-cli/keys/   .json",
      "feepayerAlias": "   ",
      "fee": "0.01"
    }
  }
}
```

### LINKED ISSUE #463: Fee Payer alias with whitespaces in name prefix and/or suffix is not considered as existing
Say you already have the Fee Payer cached with the Alias name `test`. 
If you will try to add another one with the Alias name of `  test  ` (mind whitespaces in prefix and suffix) then application won't validate it as the duplicate and will silently replace existing one.

### LINKED ISSUE #475: Add more URL, Fee and Fee Payer Alias validation cases for the `config.spec.ts` after the issues fix
- https://github.com/o1-labs/zkapp-cli/issues/428
- https://github.com/o1-labs/zkapp-cli/issues/462
- https://github.com/o1-labs/zkapp-cli/issues/463

### PR DESCRIPTION
**Pull request by shimkiv**
Closes #421
Closes #428 
Closes #462 
Closes #463 
Closes #475

### CODE REVIEW

File: src/lib/prompts.js [hunk 1]
```diff
@@ -295,4 +282,8 @@ function capitalize(string) {
   return string.charAt(0).toUpperCase() + string.slice(1);
 }
 
+function sanitizeAliasName(aliasName) {
```
Comment (ymekuria) [inline src/lib/prompts.js:285]: Good idea to refactor this into a function!

**Review by ymekuria — APPROVED** [review]
Nice! Thanks for adding the extra validation. This will improve the DX for everyone!
